typed_python/PySetInstance.cpp

```diff
@@ -93,6 +100,288 @@ PyObject* PySetInstance::setClear(PyObject* o, PyObject* args) {
     Py_RETURN_NONE;
 }
 
+void PySetInstance::copy_elements(PyObject* dst, PyObject* src) {
+    PySetInstance* dst_w = (PySetInstance*)dst;
+
+    PyObjectStealer iterator(PyObject_GetIter(src));
+    if (!iterator) {
+        throw PythonExceptionSet();
+    }
+
+    PyObject* item;
+    while ((item = PyObjectStealer(PyIter_Next(iterator)))) {
```

Comment:
passing this into a temporary PyObjectStealer like this is not a good idea.  I'm not sure what it will do. The 'stealer' pattern is intended to acknowledge that it's constructed with an incref, and needs to decref, and to use RAII to prevent exceptions from creating leaked increfs. But in this case, the PyObjectStealer is temporary, so it's just pre-decreffing the container item before storing it in 'item' which will be fine as long as the item is held by the container, but won't work if this were a generator, or something else that returns an object with only one refcount.

A safer pattern would be to use the 'iterate' function defined in util.hpp. E.g.
```
        iterate(src, [&](PyObject* item) {
            ..body of your loop here...
        });
```

We also need to check whether 'src' is a typed_python Set, Tuple, or ListOf with the same element type, and if so, bypass the python interpreter and insert the elements directly. See, for instance, `PyListOfInstance::listExtend` which checks if we are extending a list with another list of the same type, and if so, uses a much faster path.

Reply:
@braxtonmckee , thanks for feedback. took care of replacing the Stealer pattern with iterate  (simple but sweet function). Also, added fast path branching for typed python types where applicable.